### PR TITLE
[CARBONDATA-595]Fixed NPE while droping the table with HDFS lock

### PR DESCRIPTION
**Pull request by ravipesala**
Fixed NPE while droping the table with HDFS lock configured

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/467/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/508/

File: core/src/main/java/org/apache/carbondata/locks/HdfsFileLock.java [hunk 1]
```diff
@@ -108,7 +109,9 @@ public boolean unlock() {
       } catch (IOException e) {
         return false;
       } finally {
-        if (FileFactory.getCarbonFile(location, FileFactory.getFileType(location)).delete()) {
+        CarbonFile carbonFile =
+            FileFactory.getCarbonFile(location, FileFactory.getFileType(location));
+        if (carbonFile.exists() && carbonFile.delete()) {
           LOGGER.info("Deleted the lock file " + location);
         } else {
           LOGGER.error("Not able to delete the lock file " + location);
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/locks/HdfsFileLock.java]: Please handle it differently if the file is not exist. This log gives wrong indication.

**Comment by jackylk** [comment]
LGTM except one small comment

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/513/

**Comment by ravipesala** [comment]
@jackylk Fixed review comment, please review

**Pull request closed without merging**